### PR TITLE
python-single-r1.eclass: Make Py2.7 the default impl for Py2-only ebu…

### PR DESCRIPTION
**Pull request by SoapGentoo**
…ilds

### CODE REVIEW

File: eclass/python-single-r1.eclass [hunk 1]
```diff
@@ -202,7 +202,25 @@ _python_single_set_globals() {
 		optflags+=,${flags[0]/%/(+)}
 	else
 		# Multiple supported implementations; honor PYTHON_SINGLE_TARGET.
-		IUSE+=" ${flags[*]}"
+		# Check if the ebuild only supports py2 impls, if so
+		# we will later '+' to CPython 2.7
+		local py2_only=true
```
Comment (mgorny) [inline eclass/python-single-r1.eclass]: Do you see the eclass using `true` anywhere? You didn't do the research!

File: eclass/python-single-r1.eclass [hunk 2]
```diff
@@ -202,7 +202,25 @@ _python_single_set_globals() {
...
+			if python_is_python3 "${i}"; then
+				py2_only=false
+				break
+			fi
+		done
+
+		if [[ ${py2_only} == true ]]; then
+			# Only py2 impls are in _PYTHON_SUPPORTED_IMPLS, we proceed
+			# to add +python_single_target_python2_7 to IUSE
+			IUSE+=" ${flags[*]/python_single_target_python2_7/+python_single_target_python2_7}"
```
Comment (mgorny) [inline eclass/python-single-r1.eclass]: I don't like the long form here but I guess I can live with it if there's no other way ;-P.

Comment (axs-gentoo) [inline eclass/python-single-r1.eclass]: there's another way, it just involves a loop. hold on.

**Comment by SoapGentoo** [comment]
@axs-gentoo @mgorny After actually getting the grep right, I noticed that @axs-gentoo list is actually correct, only 8 ebuilds are affected. Scrap this mucking around with eclasses.

File: app-text/asciidoc/asciidoc-8.6.9-r5.ebuild [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 EAPI=6
 
-PYTHON_COMPAT=( python2_7 )
+PYTHON_COMPAT=( python2_7 pypy )
```
Comment (mgorny) [inline app-text/asciidoc/asciidoc-8.6.9-r5.ebuild]: Are you sure this is safe? Didn't I remove it because it didn't work?

Comment (marcecj) [inline app-text/asciidoc/asciidoc-8.6.9-r5.ebuild]: How did you determine that it didn't work?  The asciidoc test suite always passed when I tested with pypy.  Anyway, if you can wait, I'll test it tomorrow at the latest (by emerging packages that require asciidoc, that is, not just the test suite).

(Also, wasn't it @SoapGentoo who removed it about a week ago?  I was actually going to open a PR to add it back myself.)

Comment (SoapGentoo) [inline app-text/asciidoc/asciidoc-8.6.9-r5.ebuild]: 1) Yes I removed it, due to the python-single-r1 target thing, it's absolutely safe and passes all tests
2) and yes, I should've asked @marcecj

**Pull request closed without merging**